test/e2e/main_test.go

```diff
@@ -179,10 +148,18 @@ func TestMain(m *testing.M) {
 
 	// We want to destroy the cluster if we created it, but only if we created it,
 	// otherwise the random name will be meaningless.
-	if *destroy && isKindCluster {
-		finish = []env.Func{envfuncs.DestroyKindCluster(clusterName)}
+	if E2EConfig.ShouldDestroyKindCluster() {
+		finish = []env.Func{envfuncs.DestroyKindCluster(E2EConfig.GetKindClusterName())}
 	}
 
+	// Check that all features are specifying a suite they belong to via LabelTestSuite.
+	environment.BeforeEachFeature(func(ctx context.Context, _ *envconf.Config, t *testing.T, feature features.Feature) (context.Context, error) {
+		if _, exists := feature.Labels()[config.LabelTestSuite]; !exists {
+			t.Fatalf("Feature %s does not have a %s label, setting it to %s", feature.Name(), config.LabelTestSuite, config.TestSuiteDefault)
```

Comment:
Is the message here right? I read it as "Feature doesnt have a test suite label, so we are defaulting to the base label". But then we `t.Fatalf`?

Reply:
Ah no, that was the previous attempt at setting it by default, but it was actually not working because the feature is passed by copy and not by reference intentionally, good catch 🙏